modules/Bio/EnsEMBL/Variation/VariationFeature.pm

```diff
@@ -1968,7 +1967,117 @@ sub hgvs_genomic {
 
 }
 
+=head2  spdi_genomic
+
+Arg [1]     : int (Optional)
+              It also returns a SPDI notation for the reference allele. By default value is '0'.
+              '1' -> returns a SPDI for the reference allele;
+              '0' -> doesn't return a SPDI for the reference allele;
+Example     : my $vf = $variation_feature_adaptor->fetch_by_dbID(rs145160881);
+              my $spdi = $vf->spdi_genomic();
+              while (my ($allele,$spdi_str) = each(%{$spdi})) {
+               print "Allele $allele $spdi_str\n"; # Will print 'Allele A NC_000016.10:68684738:G:A'
+             }
+Description : Returns a reference to a hash with the alternate and reference alleles as key and a string with the genomic SPDI notation of this VariationFeature as value.
+              By default uses the slice it is placed on as reference and only returns spdi notation for the alternate alleles.
+Returntype  : Hash reference
+Exceptions  : Throws exception if VariationFeature can not be described relative to a Slice;
+              Throws exception if input value is not numeric (0 or 1)
+Caller      : general
+Status      : Experimental
+
+=cut
+
+sub spdi_genomic{
+
+  my $ref_feature = shift;
+  my $include_ref_allele = shift;
+
+  my %spdi;
+  if(!$include_ref_allele){ $include_ref_allele = 0; }
+  # If input is provided, it must have value '1' (include reference allele) or '0' (not include reference allele)
+  throw("Include reference allele must be a numeric value '1' or '0'.") unless ($include_ref_allele =~ m/^(1|0)/i);
```

Comment:
Pattern match allows any value starting 0 or 1. Case-insensitve not needed. Suggest =~ /^[01]$/ or use numerical comparison